src/main/java/org/tikv/common/operation/iterator/RawScanIterator.java

```diff
@@ -67,11 +68,18 @@ TiRegion loadCurrentRegionToCache() throws GrpcException {
     }
   }
 
-  private boolean notEndOfScan() {
-    return limit > 0
-        && !(processingLastBatch
-            && (index >= currentCache.size()
-                || Key.toRawKey(currentCache.get(index).getKey()).compareTo(endKey) >= 0));
+  private boolean endOfScan() {
+    if (limit <= 0) {
+      return true;
+    }
+    if (!processingLastBatch) {
+      return false;
+    }
+    if (index >= currentCache.size()) {
+      return true;
```

Comment:
could you add some tests to cover the reported uncovered lines?

Reply:
Delete these branches cause they will never be true.